### PR TITLE
Custom background for active item  created

### PR DESCRIPTION
**Pull request by Arcandres**
Added 2 effects to differentiate better between items, active item and hover item...
image

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -31,6 +31,10 @@
     animation: rotation 1s linear infinite;
 }
 
+.selected { 
+    background-color: #506d7e21;
+}
+
```
Comment (husseinhareb) [inline src/App.css]: Colors Cant' Be Static.

File: src/styles/sidebar-style.tsx [hunk 2]
```diff
@@ -66,10 +66,16 @@ export const List = styled.ul`
...
   margin-bottom: 10px;
   cursor: pointer;
   color: #b4b4b4;
+  padding: 5px;
+  border-radius: 5px;
+  transition-property: background;
+  transition-duration: .3s;
   &:hover {
     color: #fff;
+    background: #506d7e21;
```
Comment (husseinhareb) [inline src/styles/sidebar-style.tsx]: Colors can't be static.

**Comment by husseinhareb** [comment]
Hey, this seems like a good idea, but there are some issues with the color. As you know, the colors in this project are dynamic. This means setting the colors as

```css
.selected { 
    background-color: #506d7e21;
}
```

can be problematic, as the user might set this color as a background color or use it in a way that doesn't suit its intended design.


My approach would be using the lighten library that's been already used in the project 
as this 

```typescript
// Updated ListItem component
export const ListItem = styled.li<{ performanceSidebarBackgroundColor: string; performanceSidebarSelectedColor: string; isSelected: boolean }>`
  font-size: 14px;
  margin-bottom: 10px;
  cursor: pointer;
  padding: 5px;
  border-radius: 5px;
  transition: background-color 0.3s;
  color: ${(props) => props.isSelected ? lighten(0.1, props.performanceSidebarSelectedColor) : props.performanceSidebarSelectedColor};
  background-color: ${(props) => props.isSelected ? lighten(0.04, props.performanceSidebarBackgroundColor) : props.performanceSidebarBackgroundColor};
  transition: background-color 0.3s;

  &:hover {
    background-color: ${(props) => lighten(0.04, props.performanceSidebarBackgroundColor)};
  }
`;
```

**Comment by husseinhareb** [comment]
I pushed my changes to the pr-3 branch. Check them out and tell me what you think.


Screenshot from 2024-07-20 17-13-35

**Comment by Arcandres** [comment]
> I pushed my changes to the pr-3 branch. Check them out and tell me what you think.
> 
> Screenshot from 2024-07-20 17-13-35

Alright, I'll check it out

**Pull request closed without merging**